linera-execution/src/execution.rs

```diff
@@ -232,10 +241,16 @@ where
         };
         // Set the authenticated signer to be used in outgoing messages.
         result.authenticated_signer = signer;
-        *remaining_fuel = runtime.remaining_fuel();
+        *runtime_meter = runtime.runtime_meter();
+        if runtime_meter.bytes_read > runtime_meter.maximum_bytes_read {
+            return Err(ExecutionError::ExcessiveRead);
+        }
+        if runtime_meter.bytes_write > runtime_meter.maximum_bytes_write {
+            return Err(ExecutionError::ExcessiveWrite);
+        }
         WASM_FUEL_USED_PER_BLOCK
             .with_label_values(&[])
-            .observe((initial_remaining_fuel - *remaining_fuel) as f64);
+            .observe((initial_remaining_fuel - runtime_meter.remaining_fuel) as f64);
```

Comment:
I just realized that this is wrong: This is the used fuel per user action, but there could be multiple (or zero) user actions in each block. (@andresilva91)

Reply:
(Addressing this in https://github.com/linera-io/linera-protocol/pull/1160.)